mne/report.py

```diff
@@ -316,6 +318,62 @@ def _update_html(html, report_fname, report_sectionlabel):
     return htmls, report_fnames, report_sectionlabels
 
 
+def read_report(fname):
+    """Read a saved report that was saved as an HDF5 file.
+
+    Parameters
+    ----------
+    fname : str
+        The file name of the report to read. Must be in HDF5 format.
+
+    Returns
+    -------
+    report : instance of Report
+        The report that was read from the file.
+    """
+    report = Report()
+    report.__setstate__(read_hdf5(fname, title='mnepython'))
+    return report
+
+
+@contextmanager
+def open_report(hdf5_filename, html_filename=None, **params):
```

Comment:
I am not sure I follow the API of this. I would just put a single fname for hdf5 and rely on an `__exit__` method in Report to finish the context manager. Then you can decide to save the report.

Reply:
Now that I look at it again after a night's sleep, I agree.